### PR TITLE
Lesson 7 vlad butulan

### PR DESCRIPTION
**Pull request by vlad-o-git**
Please check my scripts for lesson 7

### CODE REVIEW

File: lesson_7/vlad_butulan/work2.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result
+
+def safe_divide(fct):
+    def check(first_parameter, second_parameter):
+        if second_parameter == 0:
+            return "Division cannot be performed"
+        else:
+            return fct(first_parameter, second_parameter)
+    return check
+
```
Comment (mariuspodean) [inline lesson_7/vlad_butulan/work2.py:10]: It is recommended that you add two spaces between functions

File: lesson_7/vlad_butulan/work2.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+print(divide(6, 3))
```
Comment (mariuspodean) [inline lesson_7/vlad_butulan/work2.py:17]: Please add a test case where you also get the message

File: lesson_7/vlad_butulan/work3.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Create a decorator called register that will update a list called print_registry with all the decorated functions names
+print_registry = []
+
+def register(fct):
+    def inner(*args):
+        print_registry.append(fct.__name__)
+        print(fct(*args))
+    return inner
```
Comment (mariuspodean) [inline lesson_7/vlad_butulan/work3.py:8]: This decorator does add the function to the registry but also does some additional stuff; I challenge you to find the bug and refactor the decorator and make it simpler.

Hint: check the returned values for the decorated functions

**Review by mariuspodean — CHANGES_REQUESTED** [review]
The code looks very good!

I've added some comments for code improvement.

File: lesson_7/vlad_butulan/work2.py (hunk 1 above)
Comment (vlad-o-git) [reply lesson_7/vlad_butulan/work2.py:10]: fixed

File: lesson_7/vlad_butulan/work2.py (hunk 2 above)
Comment (vlad-o-git) [reply lesson_7/vlad_butulan/work2.py:17]: added

File: lesson_7/vlad_butulan/work3.py (hunk 3 above)
Comment (vlad-o-git) [reply lesson_7/vlad_butulan/work3.py:8]: I see that the returned values are only the function names
Can't find the additional stuff

Comment (mariuspodean) [reply lesson_7/vlad_butulan/work3.py:8]: There is no need for the inner function here; you could have it as follows:
```python 
def register(fct):
    print_registry.append(fct.__code__.co_name)
    return fct(*args)
```

**Review by mariuspodean — APPROVED** [review]

**Pull request merged**